lib/rucio/tests/test_db.py

```diff
@@ -36,3 +38,50 @@ def test_config_poolclass():
 
     with pytest.raises(InputValidationError, match='Unknown poolclass: unknown'):
         _get_engine_poolclass('unknown')
+
+
+def test_result_to_dict_contains_all_keys(db_session, rse_factory):
```

Comment:
I always was an opponent of these low-level tests of the internal functions. Their usefulness is discutable because, on each simplest modification of the code, you have to change the tests too. And this ends with introduction of bugs both in tests an in code without anybody noticing. 

Moreover, in this particular case, you rely on models.SomeModel.__table__.columns on both sides of the assertion equality. This is almost the same as running `assert some_function() == some_function()`, which is useless.  

I would argue that we don't need these test at all. But I'll not fight at all for removing them if others consider them useful.

Reply:
Yes and no. While I see your arguments, we do test more that that. In that specific case:
- The use cases, e.g.: an empty query should return nothing, a query that returns one element should return one element. This is also a form of documentation.
- The signature of the function. This function can be seen as an interface in utils (multiple other functions call that), we therefore should test it.

And I rather have a faililng test case than not noticing, that one of the use case chages. I'd argue that the rest of our code base is not tested enough to notize _some_ changes. Some other thoughs?

Reply:
My opinion is that we must test that `did_client.get_did()` contains whatever result we want to contain. The fact that it uses your new function behind the hood automatically means your function is also tested. If tomorrow I refactor the code and use something else and not your function to generate the result for our clients, the result must not change and I must not write/change the tests for that. But it's ok to keep the test. This is a very "IMHO" comment.